includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -790,6 +809,60 @@ public function handle_return() {
 		exit;
 	}
 
+	public function log_charge_status_change( WC_Order $order, $charge ) {
+		$order->read_meta_data( true ); // Force read from db to avoid concurrent notifications
+		$old_status = $order->get_meta( 'amazon_charge_status' );
+		$charge_status = $charge->statusDetails->state; // phpcs:ignore WordPress.NamingConventions
+		if ( $charge_status === $old_status ) {
+			switch ( $old_status ) {
+				case 'AuthorizationInitiated':
+				case 'Authorized':
+				case 'CaptureInitiated':
+					wc_apa()->ipn_handler->schedule_hook( $order->get_id() );
+					break;
+			}
+			return $old_status;
+		}
+		$order->update_meta_data( 'amazon_charge_status', $charge_status );
+		$order->update_meta_data( 'amazon_charge_id', $charge->chargeId ); // phpcs:ignore WordPress.NamingConventions
+		$order->save(); // Save early for less race conditions
+
+		// @codingStandardsIgnoreStart
+		$order->add_order_note( sprintf(
+			/* translators: 1) Amazon Charge ID 2) Charge status */
+			__( 'Charge %1$s with status %2$s.', 'woocommerce-gateway-amazon-payments-advanced' ),
+			(string) $charge->chargeId,
+			(string) $charge_status
+		) );
+		// @codingStandardsIgnoreEnd
+
+		switch ( $charge_status ) {
+			case 'AuthorizationInitiated':
+			case 'Authorized':
+			case 'CaptureInitiated':
+				// Mark as on-hold.
+				$order->update_status( 'on-hold' );
+				wc_maybe_reduce_stock_levels( $order->get_id() );
+				wc_apa()->ipn_handler->schedule_hook( $order->get_id() );
+				break;
+			case 'Canceled':
+			case 'Declined':
+				$order->update_status( 'failed' );
+				wc_maybe_increase_stock_levels( $order->get_id() );
+				break;
+			case 'Captured':
```

Comment:
this event do not change order status?

Reply:
and also consider to add filters to those order status (from a ticket created) Order status shouldn't affect plugin flow and some users need to have different ones on their systems.

Reply:
> this event do not change order status?

Yes it does, payment_complete will do that.

> and also consider to add filters to those order status......

Will implement later alongside the V1 fix (if it is to be added to v1 aswell)